Prevent laptops from hibernating/suspending if lid is closed

rahul-thakoor opened the pull request:
Change-type: minor
Signed-off-by: Rahul Thakoor <rahul@balena.io>

rahul-thakoor committed:
Prevent laptops from hibernating/suspending if lid is closed

Change-type: minor
Signed-off-by: Rahul Thakoor <rahul@balena.io>

rahul-thakoor commented on power-management/Dockerfile.template:
```diff
@@ -0,0 +1,3 @@
+FROM scratch
+# copy script to create a minimal image
+COPY start.sh .
```
On device-types other than `amd64` this service does nothing

![image](https://user-images.githubusercontent.com/13220877/78703312-fed98400-791a-11ea-8295-11235fbea817.png)

chrisys commented:
@rahul-thakoor it works great preventing my laptop from sleeping but on the Jetson Nano I get the container constantly trying to install. Maybe we give it a `CMD` that does nothing and set restart to no? 

![Screenshot 2020-04-07 at 20 35 16](https://user-images.githubusercontent.com/521698/78711980-cb452c80-790f-11ea-9354-fb9668631995.png)

rahul-thakoor committed:
Use static go binary to reduce image size

- uses multi-stage build to create a static binary
- on devices that are not `amd64` the power-management service exits
- make service restart only on failure

Change-type: minor
Signed-off-by: Rahul Thakoor <rahul@balena.io>

chrisys reviewed (approved):
Super impressive optimization! Nice work @rahul-thakoor
